### DISCUSSION TITLE
reverse not showing extratags (&extratags=1), V4.0.1, import style: full. Postgresql 13, postgis30.

### DISCUSSION THREAD

**Discussion by maotm** [osm-search/Nominatim#2564] [Q&A]
Hello,
I have downloaded Nominatim v4.0.1 , installed, all OK and imported monaco-latest.osm.pbf to test.
Postgresql v13, postgis30, run import with NOMINATIM_IMPORT_STYLE=full.
I am interested in maxspeed in extratags.
Installed Nominatim UI.

Problem:  `reverse?format=json&extratags=1&lat=43.7402426&lon=7.424395&zoom=18`
will NOT output  extratags: 

```json
{
  "place_id": 101887,
  "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  "osm_type": "way",
  "osm_id": 94399829,
  "lat": "43.740267849999995",
  "lon": "7.424202699999996",
  "display_name": "Villa Europe, Boulevard Princesse Charlotte, Monte-Carlo, Monaco, 98000, Monaco",
  "address": {
    "building": "Villa Europe",
    "road": "Boulevard Princesse Charlotte",
    "suburb": "Monte-Carlo",
    "city": "Monaco",
    "postcode": "98000",
    "country": "Monaco",
    "country_code": "mc"
  },
  "extratags": {},
  "boundingbox": [
    "43.7401577",
    "43.740378",
    "7.4240514",
    "7.424354"
  ]
}
```

But extratags exist in db. Checked with UI:

```
Centre Point (lat,lon) | 43.7402426,7.424395
OSM                    | way 39838824
Place Id               | 101334 (on this server)
Computed Postcode      | 98000 (how?)
Address Tags           |  
Extra Tags             | 2 (lanes)
                       | 50 (maxspeed)
                       | yes (oneway)
                       | asphalt (surface)
```

I ran the import several times in the last 3 days, with all possible combinations of options, but could not make reverse to show maxspeed... 
Thx,
Marius

**Comment by lonvia** [comment]
You need to use the NOMINATIM_IMPORT_STYLE=extratags to get the extratags. See https://nominatim.org/release-docs/latest/admin/Import/#filtering-imported-data.

**Reply by maotm** [reply]
Hello.
I used the NOMINATIM_IMPORT_STYLE=extratags too. This is only for import, but NOMINATIM_IMPORT_STYLE=full will also import data in extratags.
The problem is not importing the extratags, which were imported correctly in both cases. The problem is that the extratags are not shown on reverse. You can try. Monaco is imported in 2 minutes.

**Reply by mtmail** [reply]
In your report you list the JSON reponse for a building (way 94399829), but in nominatim-ui output you list the response of the street (way 39838824).

https://nominatim.openstreetmap.org/reverse.php?lat=43.7402426&lon=7.424395&zoom=18&format=jsonv2&extratags=1 doesn't show extratags into the response, but changing it to street level (zoom=16) does. It is possible you forgot to change the zoom level or looked at different places?

**Reply by maotm** [reply]
Hi.
I did reverse on the same point as in nominatim-ui output (to proove my _point_ :-) ) : Centre Point (lat,lon) | 43.7402426,7.424395, but indeed, changing the zoom to 17 or 16 revealed the extratags on reverse. So a zoom level of 16 or 17 should be used. 
Thank you very much.

**Reply by jvmvl** [reply]
I included IMPORT_STYLE=extratags and it worked.
